clusterHost.py

```diff
@@ -244,6 +244,20 @@ def wait_for_masters_boot(self, desired_ip_range: Tuple[str, str]) -> None:
     def wait_for_workers_boot(self, desired_ip_range: Tuple[str, str]) -> None:
         return self._wait_for_boot(self.k8s_worker_nodes, desired_ip_range)
 
-    def teardown(self) -> None:
-        for node in self._k8s_nodes():
+    def teardown_masters(self) -> None:
+        for node in self.k8s_master_nodes:
+            node.teardown()
+
+    def teardown_workers(self) -> None:
+        for node in self.k8s_worker_nodes:
             node.teardown()
+
+    def teardown(self) -> None:
+        self.teardown_workers()
+        self.teardown_masters()
+
+    def is_master(self) -> bool:
```

Comment:
IMO a ClusterHost is just a physical host, it doesn't really make sense to mix up terms and say that it's a "master".  A clusterHost runs one or more clusterNodes.  Those can be workers or masters.

Reply:
removed from updated PR. Thanks